src/ui/filechooserwidget.cpp

```diff
@@ -0,0 +1,107 @@
+#include "filechooserwidget.h"
+
+#include <QFileDialog>
+#include <QFileInfo>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+FileChooserWidget::FileChooserWidget(QWidget* parent)
+    : QWidget(parent),
+      layout_(new QHBoxLayout(this)),
+      pathEdit_(new QLineEdit(this)),
+      mode_(Mode::Directory),
+      filter_(""),
+      openDirPath_("") {
+  _init("");
+}
+
+FileChooserWidget::FileChooserWidget(Mode mode, const QString& initialPath,
+                                     QWidget* parent)
+    : QWidget(parent),
+      layout_(new QHBoxLayout(this)),
+      pathEdit_(new QLineEdit(this)),
+      mode_(mode),
+      filter_(""),
+      openDirPath_("") {
+  _init(initialPath);
+}
+
+FileChooserWidget::FileChooserWidget(Mode mode, const QString& label,
+                                     const QString& initialPath,
+                                     QWidget* parent)
+    : QWidget(parent),
+      layout_(new QHBoxLayout(this)),
+      pathEdit_(new QLineEdit(this)),
+      mode_(mode),
+      filter_(""),
+      openDirPath_("") {
+  QLabel* lbl = new QLabel(label, this);
+  layout_->addWidget(lbl);
+  _init(initialPath);
+}
+
+void FileChooserWidget::setFileFilter(const QString& filter) {
```

Comment:
The style guide specifies Pascal case for method names.

Reply:
I didn't finish reading the guide... bloody long. I strongly hate having Pascal case method names, but alright for this project.
I forgot about this FileChooserWidget. It's reused from another project of mine. I'll update the snake_case too

Reply:
Thanks. It's not about personal preference, it's about consistency.

Reply:
sure, no pb, I completely agree